// ionic2-calendar/init-position-scroll.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from 'ionic-angular/components/scroll/scroll.ngfactory';
import * as import2 from 'ionic-angular/components/scroll/scroll';
import * as import3 from './init-position-scroll';
const styles_initPositionScrollComponent:any[] = ([] as any[]);
export const RenderType_initPositionScrollComponent:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_initPositionScrollComponent,
  data: {}
}
);
export function View_initPositionScrollComponent_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    import0.ɵqud(201326592,1,{_scrollContent: 0}),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),4,'ion-scroll',[
      [
        'scrollY',
        'true'
      ]
      ,
      [
        'style',
        'height:100%'
      ]
      ,
      [
        'zoom',
        'false'
      ]

    ]
    ,[
      [
        2,
        'scroll-x',
        (null as any)
      ]
      ,
      [
        2,
        'scroll-y',
        (null as any)
      ]

    ]
    ,(null as any),(null as any),import1.View_Scroll_0,import1.RenderType_Scroll)),
    import0.ɵdid(24576,(null as any),0,import2.Scroll,[import0.ElementRef],{
      scrollY: [
        0,
        'scrollY'
      ]
      ,
      zoom: [
        1,
        'zoom'
      ]

    }
    ,(null as any)),
    (l()(),import0.ɵted(0,['\n            '])),
    import0.ɵncd(0,0),
    (l()(),import0.ɵted(0,['\n        '])),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    const currVal_2:any = 'true';
    const currVal_3:any = 'false';
    ck(v,3,0,currVal_2,currVal_3);
  },(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,3).scrollX;
    const currVal_1:any = import0.ɵnov(v,3).scrollY;
    ck(v,2,0,currVal_0,currVal_1);
  });
}
function View_initPositionScrollComponent_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'init-position-scroll',([] as any[]),(null as any),(null as any),(null as any),View_initPositionScrollComponent_0,RenderType_initPositionScrollComponent)),
    import0.ɵdid(2449408,(null as any),0,import3.initPositionScrollComponent,[import0.ElementRef],(null as any),(null as any))
  ]
  ,(null as any),(null as any));
}
export const initPositionScrollComponentNgFactory:import0.ComponentFactory<import3.initPositionScrollComponent> = import0.ɵccf('init-position-scroll',import3.initPositionScrollComponent,View_initPositionScrollComponent_Host_0,{
  scrollX: 'scrollX',
  scrollY: 'scrollY',
  zoom: 'zoom',
  maxZoom: 'maxZoom',
  initPosition: 'initPosition',
  emitEvent: 'emitEvent'
}
,{onScroll: 'onScroll'},['*']);
