// ionic2-calendar/weekview.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from '@angular/common';
import * as import2 from 'ionic-angular/components/scroll/scroll.ngfactory';
import * as import3 from 'ionic-angular/components/scroll/scroll';
import * as import4 from './init-position-scroll.ngfactory';
import * as import5 from './init-position-scroll';
import * as import6 from './weekview';
import * as import7 from 'ionic-angular/components/slides/slides.ngfactory';
import * as import8 from 'ionic-angular/components/slides/slides';
import * as import9 from 'ionic-angular/config/config';
import * as import10 from 'ionic-angular/platform/platform';
import * as import11 from 'ionic-angular/navigation/view-controller';
import * as import12 from 'ionic-angular/components/slides/slide.ngfactory';
import * as import13 from 'ionic-angular/components/slides/slide';
import * as import14 from './calendar.service';
const styles_WeekViewComponent:any[] = ['\n        .table-fixed {\n          table-layout: fixed;\n        }\n\n        .table {\n          width: 100%;\n          max-width: 100%;\n          background-color: transparent;\n        }\n\n        .table > thead > tr > th, .table > tbody > tr > th, .table > tfoot > tr > th, .table > thead > tr > td,\n        .table > tbody > tr > td, .table > tfoot > tr > td {\n          padding: 8px;\n          line-height: 20px;\n          vertical-align: top;\n        }\n\n        .table > thead > tr > th {\n          vertical-align: bottom;\n          border-bottom: 2px solid #ddd;\n        }\n\n        .table > thead:first-child > tr:first-child > th, .table > thead:first-child > tr:first-child > td {\n          border-top: 0\n        }\n\n        .table > tbody + tbody {\n          border-top: 2px solid #ddd;\n        }\n\n        .table-bordered {\n          border: 1px solid #ddd;\n        }\n\n        .table-bordered > thead > tr > th, .table-bordered > tbody > tr > th, .table-bordered > tfoot > tr > th,\n        .table-bordered > thead > tr > td, .table-bordered > tbody > tr > td, .table-bordered > tfoot > tr > td {\n          border: 1px solid #ddd;\n        }\n\n        .table-bordered > thead > tr > th, .table-bordered > thead > tr > td {\n          border-bottom-width: 2px;\n        }\n\n        .table-striped > tbody > tr:nth-child(odd) > td, .table-striped > tbody > tr:nth-child(odd) > th {\n          background-color: #f9f9f9\n        }\n\n        .calendar-hour-column {\n          width: 50px;\n          white-space: nowrap;\n        }\n\n        .calendar-event-wrap {\n          position: relative;\n          width: 100%;\n          height: 100%;\n        }\n\n        .calendar-event {\n          position: absolute;\n          padding: 2px;\n          cursor: pointer;\n          z-index: 10000;\n        }\n\n        .calendar-cell {\n          padding: 0 !important;\n          height: 37px;\n        }\n\n        .weekview-allday-label {\n          float: left;\n          height: 100%;\n          line-height: 50px;\n          text-align: center;\n          width: 50px;\n        }\n\n        [dir="rtl"] .weekview-allday-label {\n            float: right;\n            border-right: 1px solid #ddd;\n        }\n\n        .weekview-allday-content-wrapper {\n          margin-left: 50px;\n          overflow: hidden;\n          height: 51px;\n        }\n\n        [dir="rtl"] .weekview-allday-content-wrapper {\n          margin-left: 0;\n          margin-right: 50px;\n        }\n\n        .weekview-allday-content-table {\n          min-height: 50px;\n        }\n\n        .weekview-allday-content-table td {\n          border-left: 1px solid #ddd;\n          border-right: 1px solid #ddd;\n        }\n\n        .weekview-header th {\n          overflow: hidden;\n          white-space: nowrap;\n          font-size: 14px;\n        }\n\n        .weekview-allday-table {\n          height: 50px;\n          position: relative;\n          border-bottom: 1px solid #ddd;\n          font-size: 14px;\n        }\n\n        .weekview-normal-event-container {\n          margin-top: 87px;\n          overflow: hidden;\n          left: 0;\n          right: 0;\n          top: 0;\n          bottom: 0;\n          position: absolute;\n          font-size: 14px;\n        }\n\n        .weekview .slide-zoom {\n          height: 100%;\n        }\n\n        .weekview-allday-content-wrapper scroll-content {\n          width: 100%;\n        }\n\n        ::-webkit-scrollbar,\n        *::-webkit-scrollbar {\n          display: none;\n        }\n\n        .table > tbody > tr > td.calendar-hour-column {\n          padding-left: 0;\n          padding-right: 0;\n          vertical-align: middle;\n        }\n\n        @media (max-width: 750px) {\n          .weekview-allday-label, .calendar-hour-column {\n            width: 31px;\n            font-size: 12px;\n          }\n\n          .weekview-allday-label {\n            padding-top: 4px;\n          }\n\n          .table > tbody > tr > td.calendar-hour-column {\n            padding-left: 0;\n            padding-right: 0;\n            vertical-align: middle;\n            line-height: 12px;\n          }\n\n          .table > thead > tr > th.weekview-header {\n            padding-left: 0;\n            padding-right: 0;\n            font-size: 12px;\n          }\n\n          .weekview-allday-label {\n            line-height: 20px;\n          }\n\n          .weekview-allday-content-wrapper {\n            margin-left: 31px;\n          }\n\n          [dir="rtl"] .weekview-allday-content-wrapper {\n            margin-left: 0;\n            margin-right: 31px;\n          }\n        }\n    '];
export const RenderType_WeekViewComponent:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_WeekViewComponent,
  data: {}
}
);
function View_WeekViewComponent_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'th',[[
        'class',
        'weekview-header text-center'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      '\n                        '
    ]
    ))
  ]
  ,(null as any),(ck,v) => {
    const currVal_0:any = v.context.$implicit;
    ck(v,1,0,currVal_0);
  });
}
function View_WeekViewComponent_6(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[(l()(),import0.ɵted((null as any),['\n                                                ']))],(null as any),(null as any));
}
function View_WeekViewComponent_5(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),7,'div',[
      [
        'class',
        'calendar-event'
      ]
      ,
      [
        'tappable',
        ''
      ]

    ]
      ,(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.eventSelected(v.context.$implicit.event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import1.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    import0.ɵpod([
      'top',
      'width',
      'height'
    ]
    ),
    (l()(),import0.ɵted((null as any),['\n                                                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),2,(null as any),View_WeekViewComponent_6)),
    import0.ɵdid(270336,(null as any),0,import1.NgTemplateOutlet,[import0.ViewContainerRef],{
      ngTemplateOutlet: [
        0,
        'ngTemplateOutlet'
      ]
      ,
      ngOutletContext: [
        1,
        'ngOutletContext'
      ]

    }
    ,(null as any)),
    import0.ɵpod(['displayEvent']),
    (l()(),import0.ɵted((null as any),['\n                                            ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = ck(v,2,0,((25 * v.context.$implicit.position) + 'px'),((100 * (v.context.$implicit.endIndex - v.context.$implicit.startIndex)) + '%'),'25px');
    ck(v,1,0,currVal_0);
    const currVal_1:any = co.weekviewAllDayEventTemplate;
    const currVal_2:any = ck(v,6,0,v.context.$implicit);
    ck(v,5,0,currVal_1,currVal_2);
  },(null as any));
}
function View_WeekViewComponent_4(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),8,'div',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import1.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngClass: [
        0,
        'ngClass'
      ]
    },(null as any)),
    import0.ɵpod(['calendar-event-wrap']),
    import0.ɵdid(139264,(null as any),0,import1.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    import0.ɵpod(['height']),
    (l()(),import0.ɵted((null as any),['\n                                            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_WeekViewComponent_5)),
    import0.ɵdid(401408,(null as any),0,import1.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                                        ']))
  ]
  ,(ck,v) => {
    const currVal_0:any = ck(v,2,0,(<any>v.parent).context.$implicit.events);
    ck(v,1,0,currVal_0);
    const currVal_1:any = ck(v,4,0,((25 * (<any>v.parent).context.$implicit.events.length) + 'px'));
    ck(v,3,0,currVal_1);
    const currVal_2:any = (<any>v.parent).context.$implicit.events;
    ck(v,7,0,currVal_2);
  },(null as any));
}
function View_WeekViewComponent_3(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),4,'td',[[
        'class',
        'calendar-cell'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                                        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_WeekViewComponent_4)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                                    ']))
  ]
  ,(ck,v) => {
    const currVal_0:any = v.context.$implicit.events;
    ck(v,3,0,currVal_0);
  },(null as any));
}
function View_WeekViewComponent_11(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[(l()(),import0.ɵted((null as any),['\n                                             ']))],(null as any),(null as any));
}
function View_WeekViewComponent_10(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),7,'div',[
      [
        'class',
        'calendar-event'
      ]
      ,
      [
        'tappable',
        ''
      ]

    ]
      ,(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.eventSelected(v.context.$implicit.event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import1.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    import0.ɵpod([
      'top',
      'left',
      'width',
      'height'
    ]
    ),
    (l()(),import0.ɵted((null as any),['\n                                             '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),2,(null as any),View_WeekViewComponent_11)),
    import0.ɵdid(270336,(null as any),0,import1.NgTemplateOutlet,[import0.ViewContainerRef],{
      ngTemplateOutlet: [
        0,
        'ngTemplateOutlet'
      ]
      ,
      ngOutletContext: [
        1,
        'ngOutletContext'
      ]

    }
    ,(null as any)),
    import0.ɵpod(['displayEvent']),
    (l()(),import0.ɵted((null as any),['\n                                        ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = ck(v,2,0,(((37 * v.context.$implicit.startOffset) / co.hourParts) + 'px'),(((100 / v.context.$implicit.overlapNumber) * v.context.$implicit.position) + '%'),((100 / v.context.$implicit.overlapNumber) + '%'),((37 * ((v.context.$implicit.endIndex - v.context.$implicit.startIndex) - ((v.context.$implicit.endOffset + v.context.$implicit.startOffset) / co.hourParts))) + 'px'));
    ck(v,1,0,currVal_0);
    const currVal_1:any = co.weekviewNormalEventTemplate;
    const currVal_2:any = ck(v,6,0,v.context.$implicit);
    ck(v,5,0,currVal_1,currVal_2);
  },(null as any));
}
function View_WeekViewComponent_9(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),6,'div',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import1.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngClass: [
        0,
        'ngClass'
      ]
    },(null as any)),
    import0.ɵpod(['calendar-event-wrap']),
    (l()(),import0.ɵted((null as any),['\n                                        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_WeekViewComponent_10)),
    import0.ɵdid(401408,(null as any),0,import1.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                                    ']))
  ]
  ,(ck,v) => {
    const currVal_0:any = ck(v,2,0,(<any>v.parent).context.$implicit.events);
    ck(v,1,0,currVal_0);
    const currVal_1:any = (<any>v.parent).context.$implicit.events;
    ck(v,5,0,currVal_1);
  },(null as any));
}
function View_WeekViewComponent_8(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),4,'td',[
      [
        'class',
        'calendar-cell'
      ]
      ,
      [
        'tappable',
        ''
      ]

    ]
      ,(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.select(v.context.$implicit.time,v.context.$implicit.events)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                                    '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_WeekViewComponent_9)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                                ']))
  ]
  ,(ck,v) => {
    const currVal_0:any = v.context.$implicit.events;
    ck(v,3,0,currVal_0);
  },(null as any));
}
function View_WeekViewComponent_7(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),7,'tr',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                                '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'td',[[
        'class',
        'calendar-hour-column text-center'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '\n                                    ',
      '\n                                '
    ]
    )),
    (l()(),import0.ɵted((null as any),['\n                                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_WeekViewComponent_8)),
    import0.ɵdid(401408,(null as any),0,import1.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                            ']))
  ]
  ,(ck,v) => {
    const currVal_1:any = v.context.$implicit;
    ck(v,6,0,currVal_1);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.hourColumnLabels[v.context.index];
    ck(v,3,0,currVal_0);
  });
}
function View_WeekViewComponent_2(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),36,'div',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                    '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),20,'div',[[
        'class',
        'weekview-allday-table'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                        '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'div',[[
        'class',
        'weekview-allday-label'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    )),
    (l()(),import0.ɵted((null as any),['\n                        '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),14,'ion-scroll',[
      [
        'class',
        'weekview-allday-content-wrapper'
      ]
      ,
      [
        'scrollY',
        'true'
      ]
      ,
      [
        'zoom',
        'false'
      ]

    ]
    ,[
      [
        2,
        'scroll-x',
        (null as any)
      ]
      ,
      [
        2,
        'scroll-y',
        (null as any)
      ]

    ]
    ,(null as any),(null as any),import2.View_Scroll_0,import2.RenderType_Scroll)),
    import0.ɵdid(24576,(null as any),0,import3.Scroll,[import0.ElementRef],{
      scrollY: [
        0,
        'scrollY'
      ]
      ,
      zoom: [
        1,
        'zoom'
      ]

    }
    ,(null as any)),
    (l()(),import0.ɵted(0,['\n                            '])),
      (l()(),import0.ɵeld(0,(null as any),0,10,'table',[[
        'class',
        'table table-fixed weekview-allday-content-table'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                                '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),7,'tbody',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                                '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),4,'tr',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                                    '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_WeekViewComponent_3)),
    import0.ɵdid(401408,(null as any),0,import1.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                                '])),
    (l()(),import0.ɵted((null as any),['\n                                '])),
    (l()(),import0.ɵted((null as any),['\n                            '])),
    (l()(),import0.ɵted(0,['\n                        '])),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵted((null as any),['\n                    '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),11,'init-position-scroll',[[
        'class',
        'weekview-normal-event-container'
      ]
      ],(null as any),[[
        (null as any),
        'onScroll'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('onScroll' === en)) {
        const pd_0:any = ((<any>co.setScrollPosition($event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },import4.View_initPositionScrollComponent_0,import4.RenderType_initPositionScrollComponent)),
    import0.ɵdid(2449408,(null as any),0,import5.initPositionScrollComponent,[import0.ElementRef],{
      initPosition: [
        0,
        'initPosition'
      ]
      ,
      emitEvent: [
        1,
        'emitEvent'
      ]

    }
    ,{onScroll: 'onScroll'}),
    (l()(),import0.ɵted(0,['\n                        '])),
      (l()(),import0.ɵeld(0,(null as any),0,7,'table',[[
        'class',
        'table table-bordered table-fixed weekview-normal-event-table'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                            '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),4,'tbody',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_WeekViewComponent_7)),
    import0.ɵdid(401408,(null as any),0,import1.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                            '])),
    (l()(),import0.ɵted((null as any),['\n                        '])),
    (l()(),import0.ɵted(0,['\n                    '])),
    (l()(),import0.ɵted((null as any),['\n                ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_3:any = 'true';
    const currVal_4:any = 'false';
    ck(v,8,0,currVal_3,currVal_4);
    const currVal_5:any = co.views[0].dates;
    ck(v,17,0,currVal_5);
    const currVal_6:any = co.initScrollPosition;
    const currVal_7:any = co.preserveScrollPosition;
    ck(v,25,0,currVal_6,currVal_7);
    const currVal_8:any = co.views[0].rows;
    ck(v,32,0,currVal_8);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.allDayLabel;
    ck(v,5,0,currVal_0);
    const currVal_1:any = import0.ɵnov(v,8).scrollX;
    const currVal_2:any = import0.ɵnov(v,8).scrollY;
    ck(v,7,0,currVal_1,currVal_2);
  });
}
function View_WeekViewComponent_13(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'td',[[
        'class',
        'calendar-cell'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                                    ']))
  ]
  ,(null as any),(null as any));
}
function View_WeekViewComponent_15(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'td',[[
        'class',
        'calendar-cell'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                                ']))
  ]
  ,(null as any),(null as any));
}
function View_WeekViewComponent_14(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),7,'tr',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                                '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'td',[[
        'class',
        'calendar-hour-column text-center'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '\n                                    ',
      '\n                                '
    ]
    )),
    (l()(),import0.ɵted((null as any),['\n                                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_WeekViewComponent_15)),
    import0.ɵdid(401408,(null as any),0,import1.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                            ']))
  ]
  ,(ck,v) => {
    const currVal_1:any = v.context.$implicit;
    ck(v,6,0,currVal_1);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.hourColumnLabels[v.context.index];
    ck(v,3,0,currVal_0);
  });
}
function View_WeekViewComponent_12(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),36,'div',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                    '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),20,'div',[[
        'class',
        'weekview-allday-table'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                        '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'div',[[
        'class',
        'weekview-allday-label'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    )),
    (l()(),import0.ɵted((null as any),['\n                        '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),14,'ion-scroll',[
      [
        'class',
        'weekview-allday-content-wrapper'
      ]
      ,
      [
        'scrollY',
        'true'
      ]
      ,
      [
        'zoom',
        'false'
      ]

    ]
    ,[
      [
        2,
        'scroll-x',
        (null as any)
      ]
      ,
      [
        2,
        'scroll-y',
        (null as any)
      ]

    ]
    ,(null as any),(null as any),import2.View_Scroll_0,import2.RenderType_Scroll)),
    import0.ɵdid(24576,(null as any),0,import3.Scroll,[import0.ElementRef],{
      scrollY: [
        0,
        'scrollY'
      ]
      ,
      zoom: [
        1,
        'zoom'
      ]

    }
    ,(null as any)),
    (l()(),import0.ɵted(0,['\n                            '])),
      (l()(),import0.ɵeld(0,(null as any),0,10,'table',[[
        'class',
        'table table-fixed weekview-allday-content-table'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                                '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),7,'tbody',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                                '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),4,'tr',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                                    '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_WeekViewComponent_13)),
    import0.ɵdid(401408,(null as any),0,import1.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                                '])),
    (l()(),import0.ɵted((null as any),['\n                                '])),
    (l()(),import0.ɵted((null as any),['\n                            '])),
    (l()(),import0.ɵted(0,['\n                        '])),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵted((null as any),['\n                    '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),11,'init-position-scroll',[[
        'class',
        'weekview-normal-event-container'
      ]
    ],(null as any),(null as any),(null as any),import4.View_initPositionScrollComponent_0,import4.RenderType_initPositionScrollComponent)),
      import0.ɵdid(2449408,(null as any),0,import5.initPositionScrollComponent,[import0.ElementRef],{initPosition: [
        0,
        'initPosition'
      ]
    },(null as any)),
    (l()(),import0.ɵted(0,['\n                        '])),
      (l()(),import0.ɵeld(0,(null as any),0,7,'table',[[
        'class',
        'table table-bordered table-fixed weekview-normal-event-table'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                            '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),4,'tbody',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_WeekViewComponent_14)),
    import0.ɵdid(401408,(null as any),0,import1.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                            '])),
    (l()(),import0.ɵted((null as any),['\n                        '])),
    (l()(),import0.ɵted(0,['\n                    '])),
    (l()(),import0.ɵted((null as any),['\n                ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_3:any = 'true';
    const currVal_4:any = 'false';
    ck(v,8,0,currVal_3,currVal_4);
    const currVal_5:any = co.views[0].dates;
    ck(v,17,0,currVal_5);
    const currVal_6:any = co.initScrollPosition;
    ck(v,25,0,currVal_6);
    const currVal_7:any = co.views[0].rows;
    ck(v,32,0,currVal_7);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.allDayLabel;
    ck(v,5,0,currVal_0);
    const currVal_1:any = import0.ɵnov(v,8).scrollX;
    const currVal_2:any = import0.ɵnov(v,8).scrollY;
    ck(v,7,0,currVal_1,currVal_2);
  });
}
function View_WeekViewComponent_16(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'th',[[
        'class',
        'weekview-header text-center'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      '\n                        '
    ]
    ))
  ]
  ,(null as any),(ck,v) => {
    const currVal_0:any = v.context.$implicit;
    ck(v,1,0,currVal_0);
  });
}
function View_WeekViewComponent_21(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[(l()(),import0.ɵted((null as any),['\n                                                ']))],(null as any),(null as any));
}
function View_WeekViewComponent_20(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),7,'div',[
      [
        'class',
        'calendar-event'
      ]
      ,
      [
        'tappable',
        ''
      ]

    ]
      ,(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.eventSelected(v.context.$implicit.event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import1.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    import0.ɵpod([
      'top',
      'width',
      'height'
    ]
    ),
    (l()(),import0.ɵted((null as any),['\n                                                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),2,(null as any),View_WeekViewComponent_21)),
    import0.ɵdid(270336,(null as any),0,import1.NgTemplateOutlet,[import0.ViewContainerRef],{
      ngTemplateOutlet: [
        0,
        'ngTemplateOutlet'
      ]
      ,
      ngOutletContext: [
        1,
        'ngOutletContext'
      ]

    }
    ,(null as any)),
    import0.ɵpod(['displayEvent']),
    (l()(),import0.ɵted((null as any),['\n                                            ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = ck(v,2,0,((25 * v.context.$implicit.position) + 'px'),((100 * (v.context.$implicit.endIndex - v.context.$implicit.startIndex)) + '%'),'25px');
    ck(v,1,0,currVal_0);
    const currVal_1:any = co.weekviewAllDayEventTemplate;
    const currVal_2:any = ck(v,6,0,v.context.$implicit);
    ck(v,5,0,currVal_1,currVal_2);
  },(null as any));
}
function View_WeekViewComponent_19(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),8,'div',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import1.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngClass: [
        0,
        'ngClass'
      ]
    },(null as any)),
    import0.ɵpod(['calendar-event-wrap']),
    import0.ɵdid(139264,(null as any),0,import1.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    import0.ɵpod(['height']),
    (l()(),import0.ɵted((null as any),['\n                                            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_WeekViewComponent_20)),
    import0.ɵdid(401408,(null as any),0,import1.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                                        ']))
  ]
  ,(ck,v) => {
    const currVal_0:any = ck(v,2,0,(<any>v.parent).context.$implicit.events);
    ck(v,1,0,currVal_0);
    const currVal_1:any = ck(v,4,0,((25 * (<any>v.parent).context.$implicit.events.length) + 'px'));
    ck(v,3,0,currVal_1);
    const currVal_2:any = (<any>v.parent).context.$implicit.events;
    ck(v,7,0,currVal_2);
  },(null as any));
}
function View_WeekViewComponent_18(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),4,'td',[[
        'class',
        'calendar-cell'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                                        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_WeekViewComponent_19)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                                    ']))
  ]
  ,(ck,v) => {
    const currVal_0:any = v.context.$implicit.events;
    ck(v,3,0,currVal_0);
  },(null as any));
}
function View_WeekViewComponent_26(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[(l()(),import0.ɵted((null as any),['\n                                             ']))],(null as any),(null as any));
}
function View_WeekViewComponent_25(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),7,'div',[
      [
        'class',
        'calendar-event'
      ]
      ,
      [
        'tappable',
        ''
      ]

    ]
      ,(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.eventSelected(v.context.$implicit.event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import1.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    import0.ɵpod([
      'top',
      'left',
      'width',
      'height'
    ]
    ),
    (l()(),import0.ɵted((null as any),['\n                                             '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),2,(null as any),View_WeekViewComponent_26)),
    import0.ɵdid(270336,(null as any),0,import1.NgTemplateOutlet,[import0.ViewContainerRef],{
      ngTemplateOutlet: [
        0,
        'ngTemplateOutlet'
      ]
      ,
      ngOutletContext: [
        1,
        'ngOutletContext'
      ]

    }
    ,(null as any)),
    import0.ɵpod(['displayEvent']),
    (l()(),import0.ɵted((null as any),['\n                                        ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = ck(v,2,0,(((37 * v.context.$implicit.startOffset) / co.hourParts) + 'px'),(((100 / v.context.$implicit.overlapNumber) * v.context.$implicit.position) + '%'),((100 / v.context.$implicit.overlapNumber) + '%'),((37 * ((v.context.$implicit.endIndex - v.context.$implicit.startIndex) - ((v.context.$implicit.endOffset + v.context.$implicit.startOffset) / co.hourParts))) + 'px'));
    ck(v,1,0,currVal_0);
    const currVal_1:any = co.weekviewNormalEventTemplate;
    const currVal_2:any = ck(v,6,0,v.context.$implicit);
    ck(v,5,0,currVal_1,currVal_2);
  },(null as any));
}
function View_WeekViewComponent_24(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),6,'div',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import1.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngClass: [
        0,
        'ngClass'
      ]
    },(null as any)),
    import0.ɵpod(['calendar-event-wrap']),
    (l()(),import0.ɵted((null as any),['\n                                        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_WeekViewComponent_25)),
    import0.ɵdid(401408,(null as any),0,import1.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                                    ']))
  ]
  ,(ck,v) => {
    const currVal_0:any = ck(v,2,0,(<any>v.parent).context.$implicit.events);
    ck(v,1,0,currVal_0);
    const currVal_1:any = (<any>v.parent).context.$implicit.events;
    ck(v,5,0,currVal_1);
  },(null as any));
}
function View_WeekViewComponent_23(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),4,'td',[
      [
        'class',
        'calendar-cell'
      ]
      ,
      [
        'tappable',
        ''
      ]

    ]
      ,(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.select(v.context.$implicit.time,v.context.$implicit.events)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                                    '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_WeekViewComponent_24)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                                ']))
  ]
  ,(ck,v) => {
    const currVal_0:any = v.context.$implicit.events;
    ck(v,3,0,currVal_0);
  },(null as any));
}
function View_WeekViewComponent_22(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),7,'tr',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                                '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'td',[[
        'class',
        'calendar-hour-column text-center'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '\n                                    ',
      '\n                                '
    ]
    )),
    (l()(),import0.ɵted((null as any),['\n                                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_WeekViewComponent_23)),
    import0.ɵdid(401408,(null as any),0,import1.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                            ']))
  ]
  ,(ck,v) => {
    const currVal_1:any = v.context.$implicit;
    ck(v,6,0,currVal_1);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.hourColumnLabels[v.context.index];
    ck(v,3,0,currVal_0);
  });
}
function View_WeekViewComponent_17(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),36,'div',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                    '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),20,'div',[[
        'class',
        'weekview-allday-table'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                        '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'div',[[
        'class',
        'weekview-allday-label'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    )),
    (l()(),import0.ɵted((null as any),['\n                        '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),14,'ion-scroll',[
      [
        'class',
        'weekview-allday-content-wrapper'
      ]
      ,
      [
        'scrollY',
        'true'
      ]
      ,
      [
        'zoom',
        'false'
      ]

    ]
    ,[
      [
        2,
        'scroll-x',
        (null as any)
      ]
      ,
      [
        2,
        'scroll-y',
        (null as any)
      ]

    ]
    ,(null as any),(null as any),import2.View_Scroll_0,import2.RenderType_Scroll)),
    import0.ɵdid(24576,(null as any),0,import3.Scroll,[import0.ElementRef],{
      scrollY: [
        0,
        'scrollY'
      ]
      ,
      zoom: [
        1,
        'zoom'
      ]

    }
    ,(null as any)),
    (l()(),import0.ɵted(0,['\n                            '])),
      (l()(),import0.ɵeld(0,(null as any),0,10,'table',[[
        'class',
        'table table-fixed weekview-allday-content-table'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                                '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),7,'tbody',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                                '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),4,'tr',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                                    '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_WeekViewComponent_18)),
    import0.ɵdid(401408,(null as any),0,import1.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                                '])),
    (l()(),import0.ɵted((null as any),['\n                                '])),
    (l()(),import0.ɵted((null as any),['\n                            '])),
    (l()(),import0.ɵted(0,['\n                        '])),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵted((null as any),['\n                    '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),11,'init-position-scroll',[[
        'class',
        'weekview-normal-event-container'
      ]
      ],(null as any),[[
        (null as any),
        'onScroll'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('onScroll' === en)) {
        const pd_0:any = ((<any>co.setScrollPosition($event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },import4.View_initPositionScrollComponent_0,import4.RenderType_initPositionScrollComponent)),
    import0.ɵdid(2449408,(null as any),0,import5.initPositionScrollComponent,[import0.ElementRef],{
      initPosition: [
        0,
        'initPosition'
      ]
      ,
      emitEvent: [
        1,
        'emitEvent'
      ]

    }
    ,{onScroll: 'onScroll'}),
    (l()(),import0.ɵted(0,['\n                        '])),
      (l()(),import0.ɵeld(0,(null as any),0,7,'table',[[
        'class',
        'table table-bordered table-fixed weekview-normal-event-table'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                            '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),4,'tbody',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_WeekViewComponent_22)),
    import0.ɵdid(401408,(null as any),0,import1.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                            '])),
    (l()(),import0.ɵted((null as any),['\n                        '])),
    (l()(),import0.ɵted(0,['\n                    '])),
    (l()(),import0.ɵted((null as any),['\n                ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_3:any = 'true';
    const currVal_4:any = 'false';
    ck(v,8,0,currVal_3,currVal_4);
    const currVal_5:any = co.views[1].dates;
    ck(v,17,0,currVal_5);
    const currVal_6:any = co.initScrollPosition;
    const currVal_7:any = co.preserveScrollPosition;
    ck(v,25,0,currVal_6,currVal_7);
    const currVal_8:any = co.views[1].rows;
    ck(v,32,0,currVal_8);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.allDayLabel;
    ck(v,5,0,currVal_0);
    const currVal_1:any = import0.ɵnov(v,8).scrollX;
    const currVal_2:any = import0.ɵnov(v,8).scrollY;
    ck(v,7,0,currVal_1,currVal_2);
  });
}
function View_WeekViewComponent_28(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'td',[[
        'class',
        'calendar-cell'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                                    ']))
  ]
  ,(null as any),(null as any));
}
function View_WeekViewComponent_30(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'td',[[
        'class',
        'calendar-cell'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                                ']))
  ]
  ,(null as any),(null as any));
}
function View_WeekViewComponent_29(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),7,'tr',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                                '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'td',[[
        'class',
        'calendar-hour-column text-center'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '\n                                    ',
      '\n                                '
    ]
    )),
    (l()(),import0.ɵted((null as any),['\n                                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_WeekViewComponent_30)),
    import0.ɵdid(401408,(null as any),0,import1.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                            ']))
  ]
  ,(ck,v) => {
    const currVal_1:any = v.context.$implicit;
    ck(v,6,0,currVal_1);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.hourColumnLabels[v.context.index];
    ck(v,3,0,currVal_0);
  });
}
function View_WeekViewComponent_27(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),36,'div',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                    '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),20,'div',[[
        'class',
        'weekview-allday-table'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                        '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'div',[[
        'class',
        'weekview-allday-label'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    )),
    (l()(),import0.ɵted((null as any),['\n                        '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),14,'ion-scroll',[
      [
        'class',
        'weekview-allday-content-wrapper'
      ]
      ,
      [
        'scrollY',
        'true'
      ]
      ,
      [
        'zoom',
        'false'
      ]

    ]
    ,[
      [
        2,
        'scroll-x',
        (null as any)
      ]
      ,
      [
        2,
        'scroll-y',
        (null as any)
      ]

    ]
    ,(null as any),(null as any),import2.View_Scroll_0,import2.RenderType_Scroll)),
    import0.ɵdid(24576,(null as any),0,import3.Scroll,[import0.ElementRef],{
      scrollY: [
        0,
        'scrollY'
      ]
      ,
      zoom: [
        1,
        'zoom'
      ]

    }
    ,(null as any)),
    (l()(),import0.ɵted(0,['\n                            '])),
      (l()(),import0.ɵeld(0,(null as any),0,10,'table',[[
        'class',
        'table table-fixed weekview-allday-content-table'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                                '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),7,'tbody',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                                '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),4,'tr',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                                    '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_WeekViewComponent_28)),
    import0.ɵdid(401408,(null as any),0,import1.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                                '])),
    (l()(),import0.ɵted((null as any),['\n                                '])),
    (l()(),import0.ɵted((null as any),['\n                            '])),
    (l()(),import0.ɵted(0,['\n                        '])),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵted((null as any),['\n                    '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),11,'init-position-scroll',[[
        'class',
        'weekview-normal-event-container'
      ]
    ],(null as any),(null as any),(null as any),import4.View_initPositionScrollComponent_0,import4.RenderType_initPositionScrollComponent)),
      import0.ɵdid(2449408,(null as any),0,import5.initPositionScrollComponent,[import0.ElementRef],{initPosition: [
        0,
        'initPosition'
      ]
    },(null as any)),
    (l()(),import0.ɵted(0,['\n                        '])),
      (l()(),import0.ɵeld(0,(null as any),0,7,'table',[[
        'class',
        'table table-bordered table-fixed weekview-normal-event-table'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                            '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),4,'tbody',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_WeekViewComponent_29)),
    import0.ɵdid(401408,(null as any),0,import1.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                            '])),
    (l()(),import0.ɵted((null as any),['\n                        '])),
    (l()(),import0.ɵted(0,['\n                    '])),
    (l()(),import0.ɵted((null as any),['\n                ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_3:any = 'true';
    const currVal_4:any = 'false';
    ck(v,8,0,currVal_3,currVal_4);
    const currVal_5:any = co.views[1].dates;
    ck(v,17,0,currVal_5);
    const currVal_6:any = co.initScrollPosition;
    ck(v,25,0,currVal_6);
    const currVal_7:any = co.views[1].rows;
    ck(v,32,0,currVal_7);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.allDayLabel;
    ck(v,5,0,currVal_0);
    const currVal_1:any = import0.ɵnov(v,8).scrollX;
    const currVal_2:any = import0.ɵnov(v,8).scrollY;
    ck(v,7,0,currVal_1,currVal_2);
  });
}
function View_WeekViewComponent_31(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'th',[[
        'class',
        'weekview-header text-center'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      '\n                        '
    ]
    ))
  ]
  ,(null as any),(ck,v) => {
    const currVal_0:any = v.context.$implicit;
    ck(v,1,0,currVal_0);
  });
}
function View_WeekViewComponent_36(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[(l()(),import0.ɵted((null as any),['\n                                                ']))],(null as any),(null as any));
}
function View_WeekViewComponent_35(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),7,'div',[
      [
        'class',
        'calendar-event'
      ]
      ,
      [
        'tappable',
        ''
      ]

    ]
      ,(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.eventSelected(v.context.$implicit.event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import1.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    import0.ɵpod([
      'top',
      'width',
      'height'
    ]
    ),
    (l()(),import0.ɵted((null as any),['\n                                                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),2,(null as any),View_WeekViewComponent_36)),
    import0.ɵdid(270336,(null as any),0,import1.NgTemplateOutlet,[import0.ViewContainerRef],{
      ngTemplateOutlet: [
        0,
        'ngTemplateOutlet'
      ]
      ,
      ngOutletContext: [
        1,
        'ngOutletContext'
      ]

    }
    ,(null as any)),
    import0.ɵpod(['displayEvent']),
    (l()(),import0.ɵted((null as any),['\n                                            ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = ck(v,2,0,((25 * v.context.$implicit.position) + 'px'),((100 * (v.context.$implicit.endIndex - v.context.$implicit.startIndex)) + '%'),'25px');
    ck(v,1,0,currVal_0);
    const currVal_1:any = co.weekviewAllDayEventTemplate;
    const currVal_2:any = ck(v,6,0,v.context.$implicit);
    ck(v,5,0,currVal_1,currVal_2);
  },(null as any));
}
function View_WeekViewComponent_34(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),8,'div',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import1.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngClass: [
        0,
        'ngClass'
      ]
    },(null as any)),
    import0.ɵpod(['calendar-event-wrap']),
    import0.ɵdid(139264,(null as any),0,import1.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    import0.ɵpod(['height']),
    (l()(),import0.ɵted((null as any),['\n                                            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_WeekViewComponent_35)),
    import0.ɵdid(401408,(null as any),0,import1.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                                        ']))
  ]
  ,(ck,v) => {
    const currVal_0:any = ck(v,2,0,(<any>v.parent).context.$implicit.events);
    ck(v,1,0,currVal_0);
    const currVal_1:any = ck(v,4,0,((25 * (<any>v.parent).context.$implicit.events.length) + 'px'));
    ck(v,3,0,currVal_1);
    const currVal_2:any = (<any>v.parent).context.$implicit.events;
    ck(v,7,0,currVal_2);
  },(null as any));
}
function View_WeekViewComponent_33(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),4,'td',[[
        'class',
        'calendar-cell'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                                        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_WeekViewComponent_34)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                                    ']))
  ]
  ,(ck,v) => {
    const currVal_0:any = v.context.$implicit.events;
    ck(v,3,0,currVal_0);
  },(null as any));
}
function View_WeekViewComponent_41(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[(l()(),import0.ɵted((null as any),['\n                                             ']))],(null as any),(null as any));
}
function View_WeekViewComponent_40(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),7,'div',[
      [
        'class',
        'calendar-event'
      ]
      ,
      [
        'tappable',
        ''
      ]

    ]
      ,(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.eventSelected(v.context.$implicit.event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import1.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    import0.ɵpod([
      'top',
      'left',
      'width',
      'height'
    ]
    ),
    (l()(),import0.ɵted((null as any),['\n                                             '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),2,(null as any),View_WeekViewComponent_41)),
    import0.ɵdid(270336,(null as any),0,import1.NgTemplateOutlet,[import0.ViewContainerRef],{
      ngTemplateOutlet: [
        0,
        'ngTemplateOutlet'
      ]
      ,
      ngOutletContext: [
        1,
        'ngOutletContext'
      ]

    }
    ,(null as any)),
    import0.ɵpod(['displayEvent']),
    (l()(),import0.ɵted((null as any),['\n                                        ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = ck(v,2,0,(((37 * v.context.$implicit.startOffset) / co.hourParts) + 'px'),(((100 / v.context.$implicit.overlapNumber) * v.context.$implicit.position) + '%'),((100 / v.context.$implicit.overlapNumber) + '%'),((37 * ((v.context.$implicit.endIndex - v.context.$implicit.startIndex) - ((v.context.$implicit.endOffset + v.context.$implicit.startOffset) / co.hourParts))) + 'px'));
    ck(v,1,0,currVal_0);
    const currVal_1:any = co.weekviewNormalEventTemplate;
    const currVal_2:any = ck(v,6,0,v.context.$implicit);
    ck(v,5,0,currVal_1,currVal_2);
  },(null as any));
}
function View_WeekViewComponent_39(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),6,'div',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import1.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngClass: [
        0,
        'ngClass'
      ]
    },(null as any)),
    import0.ɵpod(['calendar-event-wrap']),
    (l()(),import0.ɵted((null as any),['\n                                        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_WeekViewComponent_40)),
    import0.ɵdid(401408,(null as any),0,import1.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                                    ']))
  ]
  ,(ck,v) => {
    const currVal_0:any = ck(v,2,0,(<any>v.parent).context.$implicit.events);
    ck(v,1,0,currVal_0);
    const currVal_1:any = (<any>v.parent).context.$implicit.events;
    ck(v,5,0,currVal_1);
  },(null as any));
}
function View_WeekViewComponent_38(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),4,'td',[
      [
        'class',
        'calendar-cell'
      ]
      ,
      [
        'tappable',
        ''
      ]

    ]
      ,(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.select(v.context.$implicit.time,v.context.$implicit.events)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                                    '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_WeekViewComponent_39)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                                ']))
  ]
  ,(ck,v) => {
    const currVal_0:any = v.context.$implicit.events;
    ck(v,3,0,currVal_0);
  },(null as any));
}
function View_WeekViewComponent_37(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),7,'tr',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                                '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'td',[[
        'class',
        'calendar-hour-column text-center'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '\n                                    ',
      '\n                                '
    ]
    )),
    (l()(),import0.ɵted((null as any),['\n                                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_WeekViewComponent_38)),
    import0.ɵdid(401408,(null as any),0,import1.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                            ']))
  ]
  ,(ck,v) => {
    const currVal_1:any = v.context.$implicit;
    ck(v,6,0,currVal_1);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.hourColumnLabels[v.context.index];
    ck(v,3,0,currVal_0);
  });
}
function View_WeekViewComponent_32(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),36,'div',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                    '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),20,'div',[[
        'class',
        'weekview-allday-table'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                        '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'div',[[
        'class',
        'weekview-allday-label'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    )),
    (l()(),import0.ɵted((null as any),['\n                        '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),14,'ion-scroll',[
      [
        'class',
        'weekview-allday-content-wrapper'
      ]
      ,
      [
        'scrollY',
        'true'
      ]
      ,
      [
        'zoom',
        'false'
      ]

    ]
    ,[
      [
        2,
        'scroll-x',
        (null as any)
      ]
      ,
      [
        2,
        'scroll-y',
        (null as any)
      ]

    ]
    ,(null as any),(null as any),import2.View_Scroll_0,import2.RenderType_Scroll)),
    import0.ɵdid(24576,(null as any),0,import3.Scroll,[import0.ElementRef],{
      scrollY: [
        0,
        'scrollY'
      ]
      ,
      zoom: [
        1,
        'zoom'
      ]

    }
    ,(null as any)),
    (l()(),import0.ɵted(0,['\n                            '])),
      (l()(),import0.ɵeld(0,(null as any),0,10,'table',[[
        'class',
        'table table-fixed weekview-allday-content-table'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                                '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),7,'tbody',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                                '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),4,'tr',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                                    '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_WeekViewComponent_33)),
    import0.ɵdid(401408,(null as any),0,import1.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                                '])),
    (l()(),import0.ɵted((null as any),['\n                                '])),
    (l()(),import0.ɵted((null as any),['\n                            '])),
    (l()(),import0.ɵted(0,['\n                        '])),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵted((null as any),['\n                    '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),11,'init-position-scroll',[[
        'class',
        'weekview-normal-event-container'
      ]
      ],(null as any),[[
        (null as any),
        'onScroll'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('onScroll' === en)) {
        const pd_0:any = ((<any>co.setScrollPosition($event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },import4.View_initPositionScrollComponent_0,import4.RenderType_initPositionScrollComponent)),
    import0.ɵdid(2449408,(null as any),0,import5.initPositionScrollComponent,[import0.ElementRef],{
      initPosition: [
        0,
        'initPosition'
      ]
      ,
      emitEvent: [
        1,
        'emitEvent'
      ]

    }
    ,{onScroll: 'onScroll'}),
    (l()(),import0.ɵted(0,['\n                        '])),
      (l()(),import0.ɵeld(0,(null as any),0,7,'table',[[
        'class',
        'table table-bordered table-fixed weekview-normal-event-table'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                            '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),4,'tbody',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_WeekViewComponent_37)),
    import0.ɵdid(401408,(null as any),0,import1.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                            '])),
    (l()(),import0.ɵted((null as any),['\n                        '])),
    (l()(),import0.ɵted(0,['\n                    '])),
    (l()(),import0.ɵted((null as any),['\n                ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_3:any = 'true';
    const currVal_4:any = 'false';
    ck(v,8,0,currVal_3,currVal_4);
    const currVal_5:any = co.views[2].dates;
    ck(v,17,0,currVal_5);
    const currVal_6:any = co.initScrollPosition;
    const currVal_7:any = co.preserveScrollPosition;
    ck(v,25,0,currVal_6,currVal_7);
    const currVal_8:any = co.views[2].rows;
    ck(v,32,0,currVal_8);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.allDayLabel;
    ck(v,5,0,currVal_0);
    const currVal_1:any = import0.ɵnov(v,8).scrollX;
    const currVal_2:any = import0.ɵnov(v,8).scrollY;
    ck(v,7,0,currVal_1,currVal_2);
  });
}
function View_WeekViewComponent_43(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'td',[[
        'class',
        'calendar-cell'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                                    ']))
  ]
  ,(null as any),(null as any));
}
function View_WeekViewComponent_45(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'td',[[
        'class',
        'calendar-cell'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                                ']))
  ]
  ,(null as any),(null as any));
}
function View_WeekViewComponent_44(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),7,'tr',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                                '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'td',[[
        'class',
        'calendar-hour-column text-center'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '\n                                    ',
      '\n                                '
    ]
    )),
    (l()(),import0.ɵted((null as any),['\n                                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_WeekViewComponent_45)),
    import0.ɵdid(401408,(null as any),0,import1.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                            ']))
  ]
  ,(ck,v) => {
    const currVal_1:any = v.context.$implicit;
    ck(v,6,0,currVal_1);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.hourColumnLabels[v.context.index];
    ck(v,3,0,currVal_0);
  });
}
function View_WeekViewComponent_42(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),36,'div',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                    '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),20,'div',[[
        'class',
        'weekview-allday-table'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                        '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'div',[[
        'class',
        'weekview-allday-label'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    )),
    (l()(),import0.ɵted((null as any),['\n                        '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),14,'ion-scroll',[
      [
        'class',
        'weekview-allday-content-wrapper'
      ]
      ,
      [
        'scrollY',
        'true'
      ]
      ,
      [
        'zoom',
        'false'
      ]

    ]
    ,[
      [
        2,
        'scroll-x',
        (null as any)
      ]
      ,
      [
        2,
        'scroll-y',
        (null as any)
      ]

    ]
    ,(null as any),(null as any),import2.View_Scroll_0,import2.RenderType_Scroll)),
    import0.ɵdid(24576,(null as any),0,import3.Scroll,[import0.ElementRef],{
      scrollY: [
        0,
        'scrollY'
      ]
      ,
      zoom: [
        1,
        'zoom'
      ]

    }
    ,(null as any)),
    (l()(),import0.ɵted(0,['\n                            '])),
      (l()(),import0.ɵeld(0,(null as any),0,10,'table',[[
        'class',
        'table table-fixed weekview-allday-content-table'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                                '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),7,'tbody',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                                '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),4,'tr',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                                    '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_WeekViewComponent_43)),
    import0.ɵdid(401408,(null as any),0,import1.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                                '])),
    (l()(),import0.ɵted((null as any),['\n                                '])),
    (l()(),import0.ɵted((null as any),['\n                            '])),
    (l()(),import0.ɵted(0,['\n                        '])),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵted((null as any),['\n                    '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),11,'init-position-scroll',[[
        'class',
        'weekview-normal-event-container'
      ]
    ],(null as any),(null as any),(null as any),import4.View_initPositionScrollComponent_0,import4.RenderType_initPositionScrollComponent)),
      import0.ɵdid(2449408,(null as any),0,import5.initPositionScrollComponent,[import0.ElementRef],{initPosition: [
        0,
        'initPosition'
      ]
    },(null as any)),
    (l()(),import0.ɵted(0,['\n                        '])),
      (l()(),import0.ɵeld(0,(null as any),0,7,'table',[[
        'class',
        'table table-bordered table-fixed weekview-normal-event-table'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                            '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),4,'tbody',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_WeekViewComponent_44)),
    import0.ɵdid(401408,(null as any),0,import1.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                            '])),
    (l()(),import0.ɵted((null as any),['\n                        '])),
    (l()(),import0.ɵted(0,['\n                    '])),
    (l()(),import0.ɵted((null as any),['\n                ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_3:any = 'true';
    const currVal_4:any = 'false';
    ck(v,8,0,currVal_3,currVal_4);
    const currVal_5:any = co.views[2].dates;
    ck(v,17,0,currVal_5);
    const currVal_6:any = co.initScrollPosition;
    ck(v,25,0,currVal_6);
    const currVal_7:any = co.views[2].rows;
    ck(v,32,0,currVal_7);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.allDayLabel;
    ck(v,5,0,currVal_0);
    const currVal_1:any = import0.ɵnov(v,8).scrollX;
    const currVal_2:any = import0.ɵnov(v,8).scrollY;
    ck(v,7,0,currVal_1,currVal_2);
  });
}
export function View_WeekViewComponent_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    import0.ɵqud(201326592,1,{slider: 0}),
    (l()(),import0.ɵted((null as any),['\n        '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),74,'ion-slides',([] as any[]),(null as any),[[
        (null as any),
        'ionSlideDidChange'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:import6.WeekViewComponent = v.component;
      if (('ionSlideDidChange' === en)) {
        const pd_0:any = ((<any>co.onSlideChanged()) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },import7.View_Slides_0,import7.RenderType_Slides)),
    import0.ɵdid(614400,[
      [
        1,
        4
      ]
      ,
      [
        'weekSlider',
        4
      ]

    ]
    ,0,import8.Slides,[
      import9.Config,
      import10.Platform,
      import0.NgZone,
      [
        2,
        import11.ViewController
      ]
      ,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      loop: [
        0,
        'loop'
      ]
      ,
      dir: [
        1,
        'dir'
      ]

    }
    ,{ionSlideDidChange: 'ionSlideDidChange'}),
    (l()(),import0.ɵted(0,['\n            '])),
    (l()(),import0.ɵeld(0,(null as any),0,22,'ion-slide',([] as any[]),(null as any),(null as any),(null as any),import12.View_Slide_0,import12.RenderType_Slide)),
    import0.ɵdid(90112,(null as any),0,import13.Slide,[
      import0.ElementRef,
      import0.Renderer,
      import8.Slides
    ]
    ,(null as any),(null as any)),
    (l()(),import0.ɵted(0,['\n                '])),
      (l()(),import0.ɵeld(0,(null as any),0,12,'table',[[
        'class',
        'table table-bordered table-fixed weekview-header'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),9,'thead',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),6,'tr',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                        '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),0,'th',[[
        'class',
        'calendar-hour-column'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_WeekViewComponent_1)),
    import0.ɵdid(401408,(null as any),0,import1.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵted(0,['\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),0,1,(null as any),View_WeekViewComponent_2)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted(0,['\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),0,1,(null as any),View_WeekViewComponent_12)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted(0,['\n            '])),
    (l()(),import0.ɵted(0,['\n            '])),
    (l()(),import0.ɵeld(0,(null as any),0,22,'ion-slide',([] as any[]),(null as any),(null as any),(null as any),import12.View_Slide_0,import12.RenderType_Slide)),
    import0.ɵdid(90112,(null as any),0,import13.Slide,[
      import0.ElementRef,
      import0.Renderer,
      import8.Slides
    ]
    ,(null as any),(null as any)),
    (l()(),import0.ɵted(0,['\n                '])),
      (l()(),import0.ɵeld(0,(null as any),0,12,'table',[[
        'class',
        'table table-bordered table-fixed weekview-header'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),9,'thead',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),6,'tr',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                        '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),0,'th',[[
        'class',
        'calendar-hour-column'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_WeekViewComponent_16)),
    import0.ɵdid(401408,(null as any),0,import1.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵted(0,['\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),0,1,(null as any),View_WeekViewComponent_17)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted(0,['\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),0,1,(null as any),View_WeekViewComponent_27)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted(0,['\n            '])),
    (l()(),import0.ɵted(0,['\n            '])),
    (l()(),import0.ɵeld(0,(null as any),0,22,'ion-slide',([] as any[]),(null as any),(null as any),(null as any),import12.View_Slide_0,import12.RenderType_Slide)),
    import0.ɵdid(90112,(null as any),0,import13.Slide,[
      import0.ElementRef,
      import0.Renderer,
      import8.Slides
    ]
    ,(null as any),(null as any)),
    (l()(),import0.ɵted(0,['\n                '])),
      (l()(),import0.ɵeld(0,(null as any),0,12,'table',[[
        'class',
        'table table-bordered table-fixed weekview-header'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),9,'thead',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),6,'tr',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                        '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),0,'th',[[
        'class',
        'calendar-hour-column'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_WeekViewComponent_31)),
    import0.ɵdid(401408,(null as any),0,import1.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵted(0,['\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),0,1,(null as any),View_WeekViewComponent_32)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted(0,['\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),0,1,(null as any),View_WeekViewComponent_42)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted(0,['\n            '])),
    (l()(),import0.ɵted(0,['\n        '])),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    var co:import6.WeekViewComponent = v.component;
    const currVal_0:any = true;
    const currVal_1:any = co.dir;
    ck(v,3,0,currVal_0,currVal_1);
    const currVal_2:any = co.views[0].dayHeaders;
    ck(v,17,0,currVal_2);
    const currVal_3:any = (0 === co.currentViewIndex);
    ck(v,23,0,currVal_3);
    const currVal_4:any = (0 !== co.currentViewIndex);
    ck(v,26,0,currVal_4);
    const currVal_5:any = co.views[1].dayHeaders;
    ck(v,41,0,currVal_5);
    const currVal_6:any = (1 === co.currentViewIndex);
    ck(v,47,0,currVal_6);
    const currVal_7:any = (1 !== co.currentViewIndex);
    ck(v,50,0,currVal_7);
    const currVal_8:any = co.views[2].dayHeaders;
    ck(v,65,0,currVal_8);
    const currVal_9:any = (2 === co.currentViewIndex);
    ck(v,71,0,currVal_9);
    const currVal_10:any = (2 !== co.currentViewIndex);
    ck(v,74,0,currVal_10);
  },(null as any));
}
function View_WeekViewComponent_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'weekview',([] as any[]),[[
        2,
        'weekview',
        (null as any)
      ]
    ],(null as any),(null as any),View_WeekViewComponent_0,RenderType_WeekViewComponent)),
    import0.ɵdid(2482176,(null as any),0,import6.WeekViewComponent,[
      import14.CalendarService,
      import0.ElementRef
    ]
    ,(null as any),(null as any))
  ]
  ,(ck,v) => {
    ck(v,1,0);
  },(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,1).class;
    ck(v,0,0,currVal_0);
  });
}
export const WeekViewComponentNgFactory:import0.ComponentFactory<import6.WeekViewComponent> = import0.ɵccf('weekview',import6.WeekViewComponent,View_WeekViewComponent_Host_0,{
  weekviewAllDayEventTemplate: 'weekviewAllDayEventTemplate',
  weekviewNormalEventTemplate: 'weekviewNormalEventTemplate',
  formatWeekTitle: 'formatWeekTitle',
  formatWeekViewDayHeader: 'formatWeekViewDayHeader',
  formatHourColumn: 'formatHourColumn',
  startingDayWeek: 'startingDayWeek',
  allDayLabel: 'allDayLabel',
  hourParts: 'hourParts',
  eventSource: 'eventSource',
  markDisabled: 'markDisabled',
  locale: 'locale',
  dateFormatter: 'dateFormatter',
  dir: 'dir',
  scrollToHour: 'scrollToHour',
  preserveScrollPosition: 'preserveScrollPosition',
  lockSwipeToPrev: 'lockSwipeToPrev',
  lockSwipes: 'lockSwipes',
  startHour: 'startHour',
  endHour: 'endHour'
}
,{
  onRangeChanged: 'onRangeChanged',
  onEventSelected: 'onEventSelected',
  onTimeSelected: 'onTimeSelected',
  onTitleChanged: 'onTitleChanged'
}
,([] as any[]));
